core-plugins/src/test/java/io/cdap/plugin/batch/aggregator/DedupAggregatorUtilsTest.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package io.cdap.plugin.batch.aggregator;
+
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.etl.api.aggregation.DeduplicateAggregationDefinition;
+import io.cdap.cdap.etl.api.relational.Capability;
+import io.cdap.cdap.etl.api.relational.Engine;
+import io.cdap.cdap.etl.api.relational.Expression;
+import io.cdap.cdap.etl.api.relational.ExpressionFactory;
+import io.cdap.cdap.etl.api.relational.ExpressionFactoryType;
+import io.cdap.cdap.etl.api.relational.Relation;
+import io.cdap.cdap.etl.api.relational.RelationalTranformContext;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.Spy;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * Test for {@link DedupAggregatorUtils}
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class DedupAggregatorUtilsTest {
+
+    //Mocks used to configure tests
+
+    @Mock
+    Relation relation;
+
+    @Mock
+    private Engine engine;
+
+    @Mock
+    private ExpressionFactory<String> expressionFactory;
+
+    @Spy
+    private RelationalTranformContext relationalTranformContext;
+
+    private List<String> uniqueFields;
+    private Schema schema;
+    private ExpressionFactory<String> expressionFactoryOriginal;
+    private Relation relationOriginal;
+
+    @Before
+    public void setUp() {
+        Mockito.doReturn(engine).when(relationalTranformContext).getEngine();
+        uniqueFields = new ArrayList<>();
+        uniqueFields.add("uniqueField");
+        schema = Schema.recordOf(
+                "schema",
+                Schema.Field.of("uniqueField", Schema.nullableOf(Schema.of(Schema.Type.STRING))),
+                Schema.Field.of("selectField", Schema.nullableOf(Schema.of(Schema.Type.STRING))),
+                Schema.Field.of("filterField", Schema.nullableOf(Schema.of(Schema.Type.STRING)))
+        );
+        relationOriginal = new Relation() {
```

Comment:
It seems you can just use mocks for this. It would save you quite a bit of overrides